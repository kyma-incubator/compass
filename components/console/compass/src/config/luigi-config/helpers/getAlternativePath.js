const regex = new RegExp('/tenant/(.*?)/(.*)(?:/.*)?');

export const getAlternativePath = (tenantName, fromPath) => {
  const currentPath = fromPath || window.location.pathname;
  const match = currentPath.match(regex);
  if (match) {
    const tenant = match[1];
    const path = match[2];
    if (tenant === tenantName) {
      // the same tenant, leave path as it is
      return `${tenantName}/${path}`;
    } else {
      // other tenant, get back to context as applications or runtimes
      const contextOnlyPath = path.split('/')[0];
      return `${tenantName}/${contextOnlyPath}`;
    }
  }
  return null;
};
