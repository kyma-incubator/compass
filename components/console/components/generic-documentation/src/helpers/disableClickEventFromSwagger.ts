export function disableClickEventFromSwagger() {
  const beginOfSwaggerClickFunctionAsString = `for (var t = e.target; t && "A" !== t.tagName;)`;

  const originalEventListener = window.addEventListener;
  window.addEventListener = (
    type: string,
    fn: (...args: any[]) => any,
    options: any,
  ) => {
    if (type === 'click') {
      const fnAsString = fn.toString();
      if (fnAsString.includes(beginOfSwaggerClickFunctionAsString)) {
        return;
      }
    }
    return originalEventListener(type, fn, options);
  };
}
