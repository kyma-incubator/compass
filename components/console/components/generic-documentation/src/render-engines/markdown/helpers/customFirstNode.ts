import { Source } from '@kyma-project/documentation-component';
import { plugins } from '@kyma-project/dc-markdown-render-engine';
import { headingPrefix } from './headingPrefix';

type Header = plugins.Header;

export function customFirstNode(
  source: Source,
  toKebabCase: (str: string) => string,
): Header | undefined {
  const title: string =
    source.data && source.data.frontmatter && source.data.frontmatter.title;

  if (!title) {
    return;
  }

  return {
    title,
    id: toKebabCase(headingPrefix(source)),
    level: 'doc-title',
    source,
  };
}
