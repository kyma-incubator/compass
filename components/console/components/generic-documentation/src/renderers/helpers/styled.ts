import styled from 'styled-components';
import { ComboboxInput, Menu } from 'fundamental-react';

const typeColumnWidth = '6em';
const listItemColumnGap = '1em';
const listItemPadding = '2 * 10px';
export const ListItem = styled.span`
   {
    justify-items: start;
    display: grid;
    grid-gap: ${listItemColumnGap};
    grid-template-columns: ${typeColumnWidth} auto;
  }
`;

export const ApiTabHeader = styled.div`
   {
    display: grid;
    align-items: center;
    grid-template-columns: auto auto;
    grid-gap: 1em;
  }
`;

export const Combobox = styled(ComboboxInput)`
   {
    flex-shrink: 0;
    min-width: calc(
      ${props => props['data-max-list-chars'] + 'ch'} + ${typeColumnWidth} +
        ${listItemColumnGap} + ${listItemPadding}
    );
  }
`;

export const List = styled(Menu)`
   {
    max-height: 28em;
    overflow-y: auto;
  }
`;
