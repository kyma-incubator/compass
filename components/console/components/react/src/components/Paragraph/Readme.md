Paragraph is our basic paragraph text element. Use it when you have a lot to say. It has four sizes: `extraSmall`, `small`, `medium` (default), and `large`. It also has four font weights: `light`, `normal` (default), `medium`, and `bold`.

```jsx
<div>
  <h5>EXTRA SMALL</h5>
  <Paragraph modifiers={['extraSmall']}>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras eu lorem eget erat feugiat sagittis eu at lorem. Fusce et arcu imperdiet dolor interdum lacinia. Mauris eu dictum erat, a convallis ex. Nulla sit amet quam quis nibh facilisis fermentum non vel neque. Vestibulum placerat libero at eros porta scelerisque.
  </Paragraph>

  <h5>SMALL</h5>
  <Paragraph modifiers={['small']}>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras eu lorem eget erat feugiat sagittis eu at lorem. Fusce et arcu imperdiet dolor interdum lacinia. Mauris eu dictum erat, a convallis ex. Nulla sit amet quam quis nibh facilisis fermentum non vel neque. Vestibulum placerat libero at eros porta scelerisque.
  </Paragraph>

  <h5>DEFAULT</h5>
  <Paragraph>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras eu lorem eget erat feugiat sagittis eu at lorem. Fusce et arcu imperdiet dolor interdum lacinia. Mauris eu dictum erat, a convallis ex. Nulla sit amet quam quis nibh facilisis fermentum non vel neque. Vestibulum placerat libero at eros porta scelerisque.
  </Paragraph>

  <h5>LARGE</h5>
  <Paragraph modifiers={['large']}>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras eu lorem eget erat feugiat sagittis eu at lorem. Fusce et arcu imperdiet dolor interdum lacinia. Mauris eu dictum erat, a convallis ex. Nulla sit amet quam quis nibh facilisis fermentum non vel neque. Vestibulum placerat libero at eros porta scelerisque.
  </Paragraph>
</div>
```
