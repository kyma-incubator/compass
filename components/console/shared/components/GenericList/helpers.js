const filterEntry = (entry, query, searchProperties) => {
  if (!query) {
    return true;
  }

  if (typeof entry === 'string') {
    return (
      entry &&
      entry
        .toString()
        .toLowerCase()
        .indexOf(query.toLowerCase()) !== -1
    );
  }

  if (!Object.keys(searchProperties).length) {
    return false;
  }

  const flattenEntry = flattenProperties(entry);
  for (const property of searchProperties) {
    if (flattenEntry.hasOwnProperty(property)) {
      const value = flattenEntry[property];
      // apply to string to include numbers
      if (
        value &&
        value
          .toString()
          .toLowerCase()
          .indexOf(query.toLowerCase()) !== -1
      ) {
        return true;
      }
    }
  }
  return false;
};

const flattenProperties = (obj, prefix = '') =>
  Object.keys(obj).reduce((properties, key) => {
    const value = obj[key];
    const prefixedKey = prefix ? `${prefix}.${key}` : key;

    if (isPrimitive(value)) {
      properties[prefixedKey] = value && value.toString();
    } else if (Array.isArray(value)) {
      properties[prefixedKey] = JSON.stringify(value);
    } else {
      Object.assign(properties, flattenProperties(value, prefixedKey));
    }

    return properties;
  }, {});

const isPrimitive = type => {
  return (
    type === null || (typeof type !== 'function' && typeof type !== 'object')
  );
};

export const filterEntries = (entries, query, searchProperties) => {
  return entries.filter(entry => filterEntry(entry, query, searchProperties));
};
