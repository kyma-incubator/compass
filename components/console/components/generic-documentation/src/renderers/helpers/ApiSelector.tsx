import React, { useState } from 'react';
import { Combobox, List, ListItem } from './styled';
import { Badge } from 'fundamental-react';
import {
  odataDefinition,
  asyncApiDefinition,
  markdownDefinition,
} from '../../constants';
import { Source } from '@kyma-project/documentation-component';

function getApiNameLength(s: Source) {
  return s.data && s.data.displayName ? s.data.displayName.length : 0;
}

const BadgeForType: React.FunctionComponent<{ type: string }> = ({ type }) => {
  let badgeType: 'success' | 'warning' | 'error' | undefined;

  if (odataDefinition.possibleTypes.includes(type)) {
    badgeType = 'warning';
  }

  if (asyncApiDefinition.possibleTypes.includes(type)) {
    badgeType = 'success';
  }

  if (markdownDefinition.possibleTypes.includes(type)) {
    badgeType = 'error';
  }

  return <Badge type={badgeType}>{type}</Badge>;
};

const ApiSelector: React.FunctionComponent<{
  sources: Source[];
  onApiSelect: (api: Source) => void;
  selectedApi: Source;
}> = ({ sources, onApiSelect, selectedApi }) => {
  const [searchText, setSearchText] = useState('');

  const filteredSources = sources.filter(
    (s: Source) =>
      (s.data &&
        s.data.displayName &&
        s.data.displayName.toUpperCase().includes(searchText.toUpperCase())) ||
      s.type.includes(searchText),
  );

  function handleInputChange(e: React.ChangeEvent<HTMLInputElement>) {
    setSearchText(e.target.value);
  }

  const maxApiNameLength = filteredSources.length
    ? getApiNameLength(
        filteredSources.reduce((prev, current) =>
          getApiNameLength(current) > getApiNameLength(prev) ? current : prev,
        ),
      )
    : 0;

  return (
    <Combobox
      onClick={(e: React.MouseEvent<HTMLElement>) => {
        const a = e.target as HTMLElement; // not sure why but it's needed, thank you typescript!
        if (a.tagName === 'INPUT' || a.tagName === 'BUTTON') {
          e.stopPropagation(); // avoid closing the dropdown due to the "opening" click âˆž
        }
      }}
      data-max-list-chars={maxApiNameLength}
      menu={
        <List>
          {filteredSources.map((s: Source, id) => (
            <a
              aria-label="api-"
              href="#"
              onClick={e => onApiSelect(s)}
              className="fd-menu__item"
              key={(s.data && s.data.displayName) || s.rawContent}
            >
              <ListItem>
                <BadgeForType type={s.type} />
                {s.data && s.data.displayName}
              </ListItem>
            </a>
          ))}
        </List>
      }
      placeholder={
        (selectedApi && selectedApi.data && selectedApi.data.displayName) ||
        (selectedApi && selectedApi.type) ||
        'Select API'
      }
      inputProps={{ onChange: handleInputChange }}
    />
  );
};

export default ApiSelector;
