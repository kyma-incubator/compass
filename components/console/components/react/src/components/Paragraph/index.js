import styled from 'styled-components';
import { applyStyleModifiers } from 'styled-components-modifiers';
import { rem } from 'polished';

import {
  fontColors,
  fontSizes,
  fontStyles,
  fontWeights,
} from '../../modifiers';

const modifierConfig = {
  ...fontColors,
  ...fontSizes,
  ...fontStyles,
  ...fontWeights,
};

const Paragraph = styled.p`
  font-size: ${props => rem(props.theme.fontSizes.medium)};
  margin: 0;
  ${applyStyleModifiers(modifierConfig)};
`;

Paragraph.modifiers = modifierConfig;

export default Paragraph;
