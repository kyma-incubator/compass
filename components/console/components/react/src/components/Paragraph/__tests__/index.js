import React from 'react';
import Paragraph from '../index';
import 'jest-styled-components';

import { renderWithTheme } from 'test_setup/helpers';

describe('Paragraph Element', () => {
  it('renders correctly', () => {
    const tree = renderWithTheme(
      <Paragraph>A paragraph tag</Paragraph>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the italic modifier', () => {
    const tree = renderWithTheme(
      <Paragraph modifiers={['italic']}>A paragraph</Paragraph>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the extraSmall modifier', () => {
    const tree = renderWithTheme(
      <Paragraph modifiers={['extraSmall']}>A paragraph</Paragraph>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the small modifier', () => {
    const tree = renderWithTheme(
      <Paragraph modifiers={['small']}>A paragraph</Paragraph>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the large modifier', () => {
    const tree = renderWithTheme(
      <Paragraph modifiers={['large']}>A paragraph</Paragraph>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the fontWeightBold modifier', () => {
    const tree = renderWithTheme(
      <Paragraph modifiers={['fontWeightBold']}>A paragraph</Paragraph>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
