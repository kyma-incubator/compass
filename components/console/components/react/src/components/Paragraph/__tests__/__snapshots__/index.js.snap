// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph Element adds the extraSmall modifier 1`] = `
.c0 {
  font-size: 1rem;
  margin: 0;
  font-size: 0.75rem;
}

<p
  className="c0"
>
  A paragraph
</p>
`;

exports[`Paragraph Element adds the fontWeightBold modifier 1`] = `
.c0 {
  font-size: 1rem;
  margin: 0;
}

<p
  className="c0"
>
  A paragraph
</p>
`;

exports[`Paragraph Element adds the italic modifier 1`] = `
.c0 {
  font-size: 1rem;
  margin: 0;
}

<p
  className="c0"
>
  A paragraph
</p>
`;

exports[`Paragraph Element adds the large modifier 1`] = `
.c0 {
  font-size: 1rem;
  margin: 0;
  font-size: 1.125rem;
}

<p
  className="c0"
>
  A paragraph
</p>
`;

exports[`Paragraph Element adds the small modifier 1`] = `
.c0 {
  font-size: 1rem;
  margin: 0;
  font-size: 0.875rem;
}

<p
  className="c0"
>
  A paragraph
</p>
`;

exports[`Paragraph Element renders correctly 1`] = `
.c0 {
  font-size: 1rem;
  margin: 0;
}

<p
  className="c0"
>
  A paragraph tag
</p>
`;
